### PR TITLE
Idempotent fix

### PR DESCRIPTION
**Pull request by Dan33l**
#### Pull Request (PR) description

Currently master is not idempotent when plugin is installed.

This PR is a rework from PR #62. Adds unit tests

### CODE REVIEW

**Comment by bastelfreak** [comment]
Could you add an acceptance test for this? They are a great tool for testing idempotence.

**Comment by Dan33l** [comment]
what i know to check idempotency is like this https://github.com/voxpupuli/puppet-openvpn/blob/master/spec/acceptance/01_openvpn_server_spec.rb#L25

and so it will need a helper that install apache mysql and so on.

is it what you are think about ?

File: spec/defines/etherpad_plugin_common_spec.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
+require 'spec_helper'
+
+describe 'etherpad::plugins::common' do
+  on_supported_os.each do |os, facts|
+    context "on #{os}" do
+      let(:pre_condition) do
+        pp = <<-EOS
+        class { 'etherpad':
```
Comment (bastelfreak) [inline spec/defines/etherpad_plugin_common_spec.rb]: could you add one test with a simple `include etherpad` ?

File: manifests/plugins/common.pp [hunk 2]
```diff
@@ -4,14 +4,11 @@
   ) {
     include nodejs
     nodejs::npm { $valid_plugin_name :
-      ensure => 'present',
-      target => "${etherpad::root_dir}/",
-    }
-    ->file { $valid_plugin_name :
```
Comment (bastelfreak) [inline manifests/plugins/common.pp]: isn't the file resource needed anymore?

File: metadata.json [hunk 3]
```diff
@@ -10,7 +10,7 @@
   "dependencies": [
     {
       "name": "puppetlabs-stdlib",
-      "version_requirement": ">= 4.13.1 < 6.0.0"
+      "version_requirement": ">= 4.17.0 < 6.0.0"
```
Comment (bastelfreak) [inline metadata.json]: can you pull this into a separate PR? It isn't 100% related to the actual issue here in this PR and it would give us a dedicated entry in the CHANGELOG.md. Same for the `puppet-nodejs` dependency
